diff --git a/meson.build b/meson.build
index 39b739e..b214a56 100644
--- a/meson.build
+++ b/meson.build
@@ -70,11 +70,14 @@ glib_dep = dependency('glib-2.0', version: glib_version_info)
 gio_dep = dependency('gio-2.0', version: glib_version_info)
 gobject_dep = dependency('gobject-2.0', version: glib_version_info)
 
-#  everything else
-libsoup_dep = dependency('libsoup-3.0', required: false)
-#    fallback to libsoup2
+libsoup_abi = get_option('libsoup-abi')
+libsoup_dep = disabler()
+if ['auto', '3.0'].contains(libsoup_abi)
+  libsoup_dep = dependency('libsoup-3.0', required: libsoup_abi.contains('3.0'))
+endif
 if not libsoup_dep.found()
-  libsoup_dep = dependency('libsoup-2.4')
+#    fallback to libsoup2
+  libsoup_dep = dependency('libsoup-2.4', required: libsoup_abi.contains('2.4'))
 endif
 libxml_dep = dependency('libxml-2.0', version: '>= 2.6.0')
 libxslt_dep = dependency('libxslt', version: '>= 1.0.0')
diff --git a/meson_options.txt b/meson_options.txt
index 13fc358..5892594 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -35,3 +35,10 @@ option('with-usb-ids-path',
     value: '',
     description: 'Location of usb.ids database'
 )
+
+option('libsoup-abi',
+    type: 'combo',
+    value: 'auto',
+    choices: ['auto', '2.4', '3.0'],
+    description: 'Select libsoup ABI version'
+)
