diff --git a/src/spice-client-glib-usb-acl-helper.c b/src/spice-client-glib-usb-acl-helper.c
index 17113e6..39a80cd 100644
--- a/src/spice-client-glib-usb-acl-helper.c
+++ b/src/spice-client-glib-usb-acl-helper.c
@@ -200,6 +200,19 @@ static void check_authorization_cb(PolkitAuthority *authority,
         return;
     }
 
+    if (access(path, W_OK) != 0 || access(path, R_OK) != 0) {
+        g_autofree char *summary = NULL;
+        g_autofree char *message = NULL;
+
+        summary = g_strdup_printf("User %i has no permissions on %s", getuid(), path);
+        message = g_strdup_printf("Launch the command `sudo setfacl -m u:%s:rw %s` from the host to "
+                                  "give your user readwrite permissions on such device",
+                                  g_get_user_name(), path);
+
+        FATAL_ERROR("%s.\n%s\n", summary, message);
+        return;
+    }
+
     fprintf(stdout, "SUCCESS\n");
     fflush(stdout);
     state = STATE_WAITING_FOR_STDIN_EOF;
