diff --git a/src/qemu/qemu_process.c b/src/qemu/qemu_process.c
index 730e59eb7e..49d99c4f70 100644
--- a/src/qemu/qemu_process.c
+++ b/src/qemu/qemu_process.c
@@ -9319,7 +9319,8 @@ qemuProcessQMPNew(const char *binary,
 static int
 qemuProcessQEMULabelUniqPath(qemuProcessQMP *proc)
 {
-    /* We cannot use the security driver here, but we should not need to. */
+    if (proc->runUid == -1)
+        return 0;
     if (chown(proc->uniqDir, proc->runUid, -1) < 0) {
         virReportSystemError(errno,
                              _("Cannot chown uniq path: %1$s"),
