diff --git a/src/qemu/qemu_process.c b/src/qemu/qemu_process.c
index 6db48b0..64f8e9c 100644
--- a/src/qemu/qemu_process.c
+++ b/src/qemu/qemu_process.c
@@ -9832,6 +9832,7 @@ qemuProcessQMPNew(const char *binary,
 static int
 qemuProcessQEMULabelUniqPath(qemuProcessQMP *proc)
 {
+    return 0;
     /* We cannot use the security driver here, but we should not need to. */
     if (chown(proc->uniqDir, proc->runUid, -1) < 0) {
         virReportSystemError(errno,
