diff --git a/src/assistant/assistant.vala b/src/assistant/assistant.vala
index caaed039..7428d574 100644
--- a/src/assistant/assistant.vala
+++ b/src/assistant/assistant.vala
@@ -159,6 +159,7 @@ private ActivityProgress create_preparation_progress () {
     [GtkCallback]
     private async void on_create_button_clicked () {
         try {
+            print("Paso 1\n");
             if (express_install_row.enabled) {
                 installer_media = unattended_installer;
 
@@ -174,26 +175,28 @@ private async void on_create_button_clicked () {
                 if (!yield installer_media.prepare (progress, cancellable))
                     return;
             }
+            print("Paso 2\n");
 
             var vm_creator = installer_media.get_vm_creator ();
             var machine = yield vm_creator.create_vm (cancellable);
             machine.title = name_entry.text;
-
+            print("Paso 3\n");
             // Apply VM preferences
             var config = machine.domain.get_config (GVir.DomainXMLFlags.INACTIVE);
             config.memory = ram_row.memory;
-
+            print("Paso 4\n");
             if (config.get_class ().find_property ("current-memory") != null)
                 config.set ("current-memory", config.memory);
-
+            print("Paso 5\n");
             machine.storage_volume.resize (storage_limit_row.memory * Osinfo.KIBIBYTES,
                                            GVir.StorageVolResizeFlags.SHRINK);
+            print("Paso 6\n");
             if (firmware_row.is_uefi)
                 installer_media.set_uefi_firmware (config, true);
-
+            print("Paso 7\n");
             // save
             machine.domain.set_config (config);
-
+            print("Paso 8\n");
             progress_bar.fraction = 1.0;
 
             // Start VM
@@ -201,7 +204,9 @@ private async void on_create_button_clicked () {
                 machine.domain.start (0);
             else
                 App.app.main_window.select_item (machine);
+            print("Paso 9\n");
             vm_creator.launch_vm (machine);
+            print("Paso 10\n");
         } catch (GLib.Error error) {
             debug ("Failed to create virtual machine: %s", error.message);
 
diff --git a/src/util-app.vala b/src/util-app.vala
index 06cc38b5..dc258fa8 100644
--- a/src/util-app.vala
+++ b/src/util-app.vala
@@ -426,6 +426,7 @@ public async bool check_selinux_context_default (out string diagnosis) {
     }
 
     public async bool check_libvirt_kvm () {
+        return true;
         try {
             string standard_output;
 
